
// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package supercontractserver;

// The Supercontract service definition.
service SupercontractServer {
  // Sends a greeting
  rpc Ping (PingRequest) returns (PingReply) {}
  rpc ExecuteCall (ExecuteRequest) returns (ExecuteReturns) {}
}

// The request message containing the user's name.
message PingRequest {
  string name = 1;
}

// The response message containing the greetings
message PingReply {
  string message = 1;
}

message ExecuteRequest {
  bytes contractKey = 1;
  bytes callId = 2;
  string fileToCall = 3;
  string functionToCall = 4;
  string parameters = 5;
  bytes poExSecretDataPrefix = 6;
  uint64 scPrepayment = 7;
  uint64 smPrepayment = 8;
}

message ExecuteReturns {
  bool success = 1;
  uint64 scConsumed = 2;
  uint64 smConsumed = 3;
  bytes poExSecretData = 4;
}